extensions/amp-analytics/0.1/session-manager.js

```diff
@@ -70,6 +76,94 @@ export class SessionManager {
 
     /** @private {!Object<string, ?SessionInfoDef>} */
     this.sessions_ = map();
+
+    /** @private {!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private {!Window} */
+    this.win_ = ampdoc.win;
+
+    /** @private {!Array<!UnlistenDef>} */
+    this.unlisteners_ = [];
+
+    /** @private {?boolean} */
+    this.isFocused_ = null;
+
+    /** @private {?boolean} */
+    this.isVisible_ = null;
+
+    /** @private {boolean} */
+    this.isOpen_ = true;
+
+    this.init_();
+  }
+
+  /**
+   * Install event listeners for engaged singals.
+   * `amp-analytics` waits for ampdoc to be visible first.
+   */
+  init_() {
+    this.setInitialEngagedSignals_();
+    this.unlisten_(
+      listen(this.win_, 'focus', () => {
+        this.isFocused_ = true;
+        this.updateEngagedForSessions_();
+      })
+    );
+    this.unlisten_(
+      listen(this.win_, 'blur', () => {
+        this.isFocused_ = false;
+        this.updateEngagedForSessions_();
+      })
+    );
+    this.unlisten_(
+      listen(this.win_, 'pageshow', () => {
```

Comment:
Nit: Remove `unlisten_` since that's not what it's really doing. You can use `push` with multiple args.
```suggestion
    this.unlisteners_.push(
      listen(this.win_, 'focus', () => {
        this.isFocused_ = true;
        this.updateEngagedForSessions_();
      }),
      listen(this.win_, 'blur', () => {
        this.isFocused_ = false;
        this.updateEngagedForSessions_();
      }),
      listen(this.win_, 'pageshow', () => {
```